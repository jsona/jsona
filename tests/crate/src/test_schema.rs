snapshot_schema_parse!("schema.jsona", parse_spec);
snapshot_schema_point!(
    "schema.jsona",
    point_spec,
    "_.value.bool",
    "_.value.array",
    "_.value.object2.k1",
    "null.value",
    "object.value.k2"
);
