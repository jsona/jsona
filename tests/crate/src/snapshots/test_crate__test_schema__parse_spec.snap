---
source: tests/crate/src/test_schema.rs
expression: output
---
{
  "$defs": {
    "Object": {
      "type": "object",
      "properties": {
        "k1": {
          "type": "string"
        },
        "k2": {
          "type": "string"
        }
      }
    }
  },
  "type": "object",
  "properties": {
    "_": {
      "type": "object",
      "properties": {
        "value": {
          "description": "A value",
          "type": "object",
          "properties": {
            "null": {},
            "bool": {
              "type": "boolean",
              "default": true
            },
            "float": {
              "type": "number"
            },
            "integer": {
              "type": "integer",
              "maximum": 10
            },
            "string": {
              "type": "string",
              "enum": [
                "apple",
                "orange",
                "banana"
              ]
            },
            "array": {
              "oneOf": [
                {
                  "type": "integer"
                },
                {
                  "type": "string"
                }
              ]
            },
            "object1": {
              "$ref": "#/$defs/Object"
            },
            "object2": {
              "$ref": "#/$defs/Object"
            },
            "object3": {
              "type": "object",
              "patternProperties": {
                ".*": {
                  "type": "string"
                }
              }
            }
          },
          "required": [
            "float"
          ]
        }
      }
    },
    "null": {
      "description": "A null annotation",
      "type": "object",
      "properties": {
        "value": {
          "type": "null"
        }
      }
    },
    "bool": {
      "description": "A bool annotation",
      "type": "object",
      "properties": {
        "value": {
          "type": "boolean"
        }
      }
    },
    "object": {
      "description": "A object annotation",
      "type": "object",
      "properties": {
        "value": {
          "type": "object",
          "properties": {
            "k1": {
              "type": "string"
            },
            "k2": {
              "type": "string"
            }
          }
        }
      }
    }
  }
}
